framework/src/modules/nft/endpoint.ts

```diff
@@ -153,48 +153,64 @@ export class NFTEndpoint extends BaseEndpoint {
 			return { collectionIDs: [] };
 		}
 
+		const isNFTSupported = await this.isNFTSupported(context);
+
+		if (!isNFTSupported.isNFTSupported) {
+			throw new Error('NFT is not supported');
+		}
+
 		const supportedNFTsData = await supportedNFTsStore.get(
 			context.getImmutableMethodContext(),
 			chainID,
 		);
 
-		return {
-			collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
-				collection.collectionID.toString('hex'),
-			),
-		};
+		if (supportedNFTsData.supportedCollectionIDArray.length) {
+			return {
+				collectionIDs: supportedNFTsData.supportedCollectionIDArray.map(collection =>
+					collection.collectionID.toString('hex'),
+				),
+			};
+		}
+
+		return { collectionIDs: [] };
 	}
 
-	public async collectionExists(
+	public async isCollectionIDSupported(
 		context: ModuleEndpointContext,
-	): Promise<{ collectionExists: boolean }> {
+	): Promise<{ isCollectionIDSupported: boolean }> {
 		const { params } = context;
 
 		validator.validate<{ chainID: string; collectionID: string }>(
-			collectionExistsRequestSchema,
+			isCollectionIDSupportedRequestSchema,
 			params,
 		);
 
 		const chainID = Buffer.from(params.chainID, 'hex');
 
+		const collectionID = Buffer.from(params.collectionID, 'hex');
+
 		const supportedNFTsStore = this.stores.get(SupportedNFTsStore);
 
 		const chainExists = await supportedNFTsStore.has(context.getImmutableMethodContext(), chainID);
 
 		if (!chainExists) {
-			return { collectionExists: false };
+			return { isCollectionIDSupported: false };
 		}
 
-		const collectionID = Buffer.from(params.collectionID, 'hex');
+		const isNFTSupported = await this.isNFTSupported(context);
+
+		if (!isNFTSupported.isNFTSupported) {
+			throw new Error('NFT is not supported');
+		}
```

Comment:
```suggestion

```

Reply:
What is the change here? The code looks the same to me 🙂

Reply:
Ahh ok. So basically at this point `isNFTSupported ` method (which is part of the NFT module) is not useful/efficient when it's entire purpose is to check if an NFT is supported? IMHO, using it's functionality prevents code duplication. But if it's more productive to remove it, am all for it 🙂

Reply:
Perhaps a new method can be added to NFTMethod that verifies if a collection is supported and can be used within `isNFTSupported` and `isCollectionSupported` endpoints, this way the additional parameter can be avoided, if code re-usability is intended.

Reply:
This would be ok but the method would be almost a duplicate of the existing one (`isNFTSupported`) in the endpoints. It uses the already defined `isNFTSupported ` in the `NFTMethod` class. Adding another `isNFTSupported` method would be duplicating functionality...I think.